test/Constraints/result_builder_diags.swift

```diff
@@ -781,3 +781,21 @@ func test_rdar65667992() {
     }
   }
 }
+
+func test_weak_with_nonoptional_type() {
+  class X {
+    func test() -> Int { 0 }
+  }
+
+  tuplify(true) { c in
+    // FIXME(diagnostic): The warning is incorrect, because `weak` cannot be declared as `let`.
+    weak var x: X = X() // expected-error {{'weak' variable should have optional type 'X?'}}
+    // expected-warning@-1 {{variable 'x' was never mutated; consider changing to 'let' constant}}
```

Comment:
@hborla I'd looking into why structural diagnostics are run for invalid result builder bodies separately.